Source/ACE/Entity/Player.cs

```diff
@@ -1605,5 +1605,30 @@ public void ClearDelayedTeleport()
             DelayedTeleportTime = DateTime.MinValue;
             WaitingForDelayedTeleport = false;
         }
+
+        override public void GameLoopUpdate()
+        {
+            uint oldHealth = Health.Current;
```

Comment:
I was looking through the code base, and I agree with this overall comment of managing how things are ticked/updated.

Right now there is a large amount of concurrency in the code (parallel landblock based/movement based object updates, and network based events).  This code-base is almost certain to develop considerable data-races and other concurrency errors without a good, standardized, encapsulated object update mechanism, such as "Updatable" objects inserted into "Update" queues (with most synchronization either managed by single threaded queues, or implicit locking via the queue handler).

I didn't want to take off too large a bite for my first commit, but its something I would be happy to look into in the future (although I'd imagine this would want further discussion via an issue request?).